src/cap32.cpp

```diff
@@ -1568,26 +1568,31 @@ void update_cpc_speed()
 
 std::string getConfigurationFilename(bool forWrite)
 {
-  // First look in any user supplied configuration file path
-  std::string configFilename = args.cfgFilePath;
-  if(access(configFilename.c_str(), F_OK) != 0) {
-     // If not found, cap32.cfg in the same directory as the executable
-     configFilename = std::string(chAppPath) + "/cap32.cfg";
-     // If not found, look for .cap32.cfg in the home of current user
-     if (access(configFilename.c_str(), F_OK) != 0) {
-        configFilename = std::string(getenv("HOME")) + "/.cap32.cfg";
-        // If still not found, look for cap32.cfg in /etc
-        if (!forWrite && access(configFilename.c_str(), F_OK) != 0) {
-           configFilename = "/etc/cap32.cfg";
-        }
-     }
-  }
-  std::cout << "Using configuration file" << (forWrite ? " to save" : "") << ": " << configFilename << std::endl;
-  return configFilename;
+   int mode = R_OK | ( F_OK * forWrite );
+
+   // If not found, look for .cap32.cfg in the XDG_CONFIG_HOME of current user 
+   // If not found, look for .cap32.cfg in the default XDG_CONFIG_HOME of current user
+   std::string defaultConfig = ( getenv("XDG_CONFIG_HOME") != nullptr ) ? \
+      std::string(getenv("XDG_CONFIG_HOME")) : (std::string(getenv("HOME")) + "/.config") + "/cap32.cfg";
```

Comment:
I'm not sure we can rely on HOME always being defined (typically on windows, but not only).
I realize previous code did, I suppose it was OK because it was always finding it locally.

I tried to think of multiple options, maybe the simplest would be something like that:

std::vector<std::pair<char*, std::string>> configPaths = {
  { "", args.cfgFilePath},
  { chAppPath, "/cap32.cfg" },
  { getenv("XDG_CONFIG_HOME"), "/cap32.cfg" },
  ...
}

for(const auto& p: configFilename){
  // Skip paths using getenv if it returned NULL (i.e environment variable not defined)
  if (!p.first) continue;
  std::string s = std::string(p.first) + p.second;
  if (access(s.c_str(), mode) == 0) 
￼    std::cout << "Using configuration file" << (forWrite ? " to save" : "") << ": " << s << std::endl;
    return s;
  }
￼}

Reply:
Sorry but i read a bit about such issue (multiplateform default config file) before i wrote this PR and i can tell that's not small code like our that solves every cases. People want better ? they fix it. I've no much time to spend on this. I solve my bugs for nixos and [Magick2cpc](https://github.com/bignaux/Magick2CPC).

Reply:
i'm reading http://koshmaar.linuxpl.info/sdl_config/news.php

Reply:
Well, solving every cases is one thing. Here we're talking about making Caprice32 segfault systematically at startup on windows. Unless I miss something, previous code was working because it was never executing the getenv when running on windows because it takes the config file from the current directory before. But here it will execute the getenv no matter what resulting in a segfault.

Unfortunately, the windows CI on appveyor doesn't execute any e2e test. I should work on integrating it. But I suspect this would fail.

Reply:
we could have #ifdef for different plateform default path settings. btw i'd like your feedback on sdl_config, if you're not against, i will try something, since it has substitution that was missing for my second part.

Reply:
I try to figure out the interest of sdl-config depending on sdl vs another good ini parsing library.

Reply:
I'm not sure which benefits this would bring. It seems to handle parsing only so wouldn't solve our current issue of finding the config file. It would add a dependency, increase compile time and makefile complexity ... The last update on the website is from 2008 and it looks like this was imported as-is on github 8 months ago and untouched since.

Reply:
benefits : reduce code coverage, share code with other produts on configuration issue, add substitution etc. if you're not ready to increase your dependancies, you're not living in my world (no cost for a git module) yes ,last update is 2008 seems dead, but github was upload 8 months ago (people from SF) and i've already made an issue to see activity. As i did here .. answere here ? 6 days, so i'll look sdl-config that way. I first do a small contribution or legitimate question, then i wait. You still doesn't have merge my contribution, let's see , perharps sdl-config has not a lot of public interesting but i could make things there, i'm not worried. Free software died because of bad dictatorship not knowing on how deal with new contributor. I've plenty of non commenting PR on project, i know now their people are not able to manage a project (navit case). I PR merged i 'll do a new improvement, no news, i do nothing. My rule.

Reply:
The configuration parsing/generation is 150 lines of code (.h and .cpp), didn't change much since its inception and never caused any issue so I'm not sure it's worth the overhead. A module that handles finding the configuration file too would be another story, but just parsing I'd advise against putting any effort into it.

Reply:
plus this not so great ones functions from cap32.cpp :
saveConfiguration() 67l
loadConfiguration() 114l
and variables substitutions in configuration would be great... btw, if you're not interessting by, no matter. Let write your test and i'll fix after failure. That way, we ensure a bit your test and my fix.

Reply:
Those methods would remain if you were to use SDL_Config, just replacing (set|get).*Value with the SDL_Config equivalent. If not, then I'm missing something.

Appveyor build is updated to run the e2e_test too so you can sync your repository to see what it says.

Reply:
Can you do a git pull and update your pull request to see what this gives on appveyor?